src/pooling.jl

```diff
@@ -136,3 +136,20 @@ if is_nnpack_available()
         return func(x, pdims; kwargs...)
     end
 end
+
+expand(N, i::Tuple) = i
+expand(N, i::Integer) = ntuple(_ -> i, N)
+
+function maxpool(x, k::NTuple{N, Integer}; pad = 0, stride = k) where N
+    pad = expand(Val(2*N), pad)
```

Comment:
Why is `pad` twice as long here? I'd expect padding to only apply to the dimensions you actually pool.

Reply:
Yeah I actually had pad the same length as the kernel. But then I looked at this implementation of MaxPool in Flux https://github.com/FluxML/Flux.jl/blob/8ee6af1bee39bd9f6ed8e6cdf854ea9aaccb14dd/src/layers/conv.jl#L287. Here it is twice the length. Should I make it the same as length of kernel?

Reply:
@staticfloat do you know why this is?

Reply:
I tried with 2*N and N separately. In both cases, the output size is the same.

Reply:
I believe this is `2 * N` for supporting asymmetric padding. If kernel is of length 2, then for 2 dimensions we will have to specify padding along 4 directions.

Reply:
Ah, that makes sense.

Assuming we can just pass `N` in for symmetric padding, and let that be handled later on, that's better here.

Reply:
Yep; it should all get canonicalized to `2*N` later on, once the `PoolDims` object is created.

Reply:
@thebhatman As per the discussion, it makes more sense to pass `N` here.